nilearn/surface/surface.py

```diff
@@ -0,0 +1,643 @@
+    row_indices = row_indices[~masked]
+    sample_indices = sample_indices[~masked]
+    weights = np.ones(len(row_indices))
+    proj = sparse.csr_matrix(
+        (weights, (row_indices, sample_indices.ravel())),
+        shape=(n_vertices, np.prod(img_shape)))
+    proj = sklearn.preprocessing.normalize(proj, axis=1, norm='l1')
+    return proj
+
+
+def _nearest_voxel_sampling(images, mesh, affine, kind='ball', radius=3.,
+                            n_points=None, mask=None):
+    """In each image, measure the intensity at each node of the mesh.
+
+    Image intensity at each sample point is that of the nearest voxel.
+    A 2-d array is returned, where each row corresponds to an image and each
+    column to a mesh vertex.
+    See documentation of vol_to_surf for details.
+
+    """
+    proj = _projection_matrix(
+        mesh, affine, images[0].shape, kind=kind, radius=radius,
+        n_points=n_points, mask=mask)
+    data = np.asarray(images).reshape(len(images), -1).T
+    texture = proj.dot(data)
+    # if all samples around a mesh vertex are outside the image,
+    # there is no reasonable value to assign to this vertex.
+    # in this case we return NaN for this vertex.
+    texture[np.asarray(proj.sum(axis=1) == 0).ravel()] = np.nan
+    return texture.T
+
+
+def _interpolation_sampling(images, mesh, affine, kind='ball', radius=3,
+                            n_points=None, mask=None):
+    """In each image, measure the intensity at each node of the mesh.
+
+    Image intensity at each sample point is computed with trilinear
+    interpolation.
+    A 2-d array is returned, where each row corresponds to an image and each
+    column to a mesh vertex.
+    See documentation of vol_to_surf for details.
+
+    """
+    sample_locations = _sample_locations(
+        mesh, affine, kind=kind, radius=radius, n_points=n_points)
+    n_vertices, n_points, img_dim = sample_locations.shape
+    grid = [np.arange(size) for size in images[0].shape]
+    interp_locations = np.vstack(sample_locations)
+    masked = _masked_indices(interp_locations, images[0].shape, mask=mask)
+    # loop over images rather than building a big array to use less memory
+    all_samples = []
+    for img in images:
+        interpolator = interpolate.RegularGridInterpolator(
+            grid, img,
+            bounds_error=False, method='linear', fill_value=None)
+        samples = interpolator(interp_locations)
+        # if all samples around a mesh vertex are outside the image,
+        # there is no reasonable value to assign to this vertex.
+        # in this case we return NaN for this vertex.
+        samples[masked] = np.nan
+        all_samples.append(samples)
+    all_samples = np.asarray(all_samples)
+    all_samples = all_samples.reshape((len(images), n_vertices, n_points))
+    texture = np.nanmean(all_samples, axis=2)
+    return texture
+
+
+def vol_to_surf(img, surf_mesh,
+                radius=3., kind='line', interpolation='nearest',
```

Comment:
Maybe I missed the discussion, but does setting the interpolation to "nearest" by default affect negatively the quality of the results?

Reply:
linear (trilinear) is available from 0.14 scipy.

Reply:
yes the results are more stable when resampling and applying affine transforms to the input 3d image when using linear interpolation rather than nearest. I put nearest by default because users of scipy 0.13 who try to use linear get a warning saying that we are falling back to nearest